lstchain/tools/lstchain_create_drs4_pedestal_file.py

```diff
@@ -0,0 +1,109 @@
+"""
+Create drs4 time pedestal fits file.
+"""
+import numpy as np
+from astropy.io import fits
+
+from ctapipe.core import Provenance, traits
+from ctapipe.core import Tool
+from ctapipe.io import EventSource
+from lstchain.calib.camera.r0 import LSTR0Corrections
+from lstchain.calib.camera.drs4 import DragonPedestal
+
+
+class PedestalFITSWriter(Tool):
+
+    name = "PedestalFITSWriter"
+    description = "Generate a pedestal FITS file"
+
+    output_file = traits.Path(
+        default_value="pedestal.fits",
+        help="Path to the generated FITS pedestal file",
+        directory_ok=False,
+        exists=False,
+    ).tag(config=True)
+
+    deltaT = traits.Bool(
+        help="Flag to use deltaT correction. Default=True", default_value=True
+    ).tag(config=True)
+
+    aliases = {
+        "input_file": "EventSource.input_url",
+        "output_file": "PedestalFITSWriter.output_file",
+    }
+
+    flags = {
+        "deltaT": ({"PedestalFITSWriter": {"deltaT": True}}, "Flag to use deltaT correction. Default is True")
+    }
+
+    classes = [EventSource, LSTR0Corrections, DragonPedestal]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a pedestal FITS file for low level calibration.
+
+        For getting help run:
+        lstchain_create_drs4_pedestal_file --help
+        """
+
+        self.eventsource = None
+        self.pixel_ids = None
+        self.pedestal = None
+        self.lst_r0 = None
+
+    def setup(self):
+
+        self.log.debug("Opening file")
+        self.eventsource = EventSource.from_config(parent=self)
+        self.lst_r0 = self.add_component(LSTR0Corrections(parent=self))
+
+        for event in self.eventsource:
```

Comment:
Setup is thought to be a `setup` function, I don't think you should be looping through all events already in `setup`, better move to start.